src/pages/[quizId]/index.tsx

```diff
@@ -8,74 +15,99 @@ import { useRouter } from 'next/router';
 import { trpc } from '../../utils/trcp';
 import { useQuizDataStore } from '../../state/quiz-data.state';
 
-const Welcome = (): JSX.Element => {
+enum FormError {
+  UNDEFINED = 'undefined',
+  TOO_SHORT = 'tooShort',
+  TOO_LONG = 'tooLong',
+}
+
+const ERROR_MESSAGES = {
+  undefined: "Please choose your team's name",
+  tooShort: "The team's name must be atleast 2 characters long",
+  tooLong: "The team's name can't be more than 20 characters long",
+};
+
+const MIN_TEAM_NAME_LENGTH = 2;
+const MAX_TEAM_NAME_LENGTH = 20;
+
+const Quiz = () => {
   const router = useRouter();
 
   const [teamName, setTeamName] = useState<string>(``);
-  const [isButtonDisabled, setIsButtonDisabled] = useState<boolean>(true);
-  const [errorMessage, setErrorMessage] = useState<string>(`Enter your name`);
+
+  const [error, setError] = useState<FormError | undefined>();
+
+  const handleTeamNameChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setTeamName(e.target.value);
+
+    if (error) setError(undefined);
+  };
 
   const { id } = useQuizDataStore((state) => state.quizData);
 
   const { mutate: createTeam } = trpc.team.createTeam.useMutation({
     onSuccess: () => null,
   });
 
-  useEffect(() => {
-    const isTeamNameInvalid = teamName.length < 2 || teamName.length > 32;
+  const handleSubmit = async () => {
+    if (!teamName) {
+      setError(FormError.UNDEFINED);
+      return;
+    }
 
-    setIsButtonDisabled(isTeamNameInvalid);
+    if (teamName.length < MIN_TEAM_NAME_LENGTH) {
+      setError(FormError.TOO_SHORT);
+      return;
+    }
 
-    if (isTeamNameInvalid) {
-      setErrorMessage(`Pleaaaase enter your team name`);
+    if (teamName.length > MAX_TEAM_NAME_LENGTH) {
+      setError(FormError.TOO_LONG);
+      return;
     }
-  }, [teamName, isButtonDisabled]);
 
-  async function onSubmitHandler() {
     if (!id) {
       throw new Error('Quiz has not been created');
     }
 
-    await createTeam({
-      name: teamName,
-      quizId: id,
-    });
+    // FIXME quizId or id? Shouldnt team just have an id? Or is that to bind it to a quiz? Then it should have both
```

Comment:
team is bind to quiz, that's why here is quizId, id for team will be created automatically by DB